packages/next/src/next-devtools/userspace/app/segment-explorer-node.tsx

```diff
@@ -0,0 +1,130 @@
+'use client'
+
+import type { ReactNode } from 'react'
+import { useState, createContext, useContext, use, useMemo } from 'react'
+import { useEffect } from 'react'
+import { dispatcher } from 'next/dist/compiled/next-devtools'
+import { notFound } from '../../../client/components/not-found'
+
+export type SegmentNodeState = {
+  type: string
+  pagePath: string
+  boundaryType: string | null
+  setBoundaryType: (type: 'error' | 'not-found' | 'loading' | null) => void
+}
+
+function SegmentTrieNode({
+  type,
+  pagePath,
+}: {
+  type: string
+  pagePath: string
+}): React.ReactNode {
+  const { boundaryType, setBoundaryType } = useSegmentState()
+  const nodeState: SegmentNodeState = useMemo(
+    () => ({
+      type,
+      pagePath,
+      boundaryType,
+      setBoundaryType,
+    }),
+    [type, pagePath, boundaryType, setBoundaryType]
+  )
+
+  useEffect(() => {
+    dispatcher.segmentExplorerNodeAdd(nodeState)
+    return () => {
+      dispatcher.segmentExplorerNodeRemove(nodeState)
+    }
+  }, [nodeState])
+
+  return null
+}
+
+function NotFoundSegmentNode() {
+  useEffect(() => {
+    notFound()
+  }, [])
+  return null
+}
+
+function ErrorSegmentNode() {
+  useEffect(() => {
+    throw new Error('__NEXT_DEVTOOLS_SEGMENT_ERROR__')
```

Comment:
```suggestion
    throw new Error('NEXT_DEVTOOLS_SIMULATED_ERROR')
```

The error is potentially visible so it doesn't need to look private. The "simulated" makes it clearer it's a synthetic error. "segment error" sounds like something with the Segment Explorer is wrong.